### PR TITLE
Backport more Comments blocks unit tests

### PR DESCRIPTION
**Pull request by ockham**
I rebased @DAreRodz' #2649 since a few of its changes have been [merged](https://github.com/WordPress/wordpress-develop/pull/2649#issuecomment-1113949397) into `trunk` (https://github.com/WordPress/wordpress-develop/commit/80fd6c30f6b5df690731275e973caa7c19bf4e53).

See https://github.com/WordPress/wordpress-develop/pull/2649#issuecomment-1114874483 for more context.

Trac ticket: https://core.trac.wordpress.org/ticket/55643

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by ockham** [comment]
cc/ @adamziel and @gziolo for possible inclusion in RC1.

**Comment by gziolo** [comment]
> cc/ @adamziel and @gziolo for possible inclusion in RC1.

We can add tests also after RC1. I added it to the project board 👍🏻

**Comment by hellofromtonya** [comment]
This PR along with unraveling why there's test-to-test interdependencies in the associated test class are both on my tomorrow TODO list.

**Comment by hellofromtonya** [comment]
This PR along with unraveling why there's test-to-test interdependencies in the associated test class are both on my tomorrow TODO list.

File: tests/phpunit/tests/blocks/renderCommentTemplate.php [hunk 1]
```diff
@@ -348,4 +378,53 @@ function test_build_comment_query_vars_from_block_with_comment_preview() {
...
+			str_replace( array( "\n", "\t" ), '', $block->render() )
+		);
+
+		remove_filter( 'wp_get_current_commenter', $commenter_filter );
+
+		// Test it again and ensure the unmoderated comment doesn't leak out.
+		$this->assertSame(
+			'<ol class="wp-block-comment-template"><li id="comment-' . self::$comment_ids[0] . '" class="comment even thread-odd thread-alt depth-1"><div class="wp-block-comment-author-name"><a rel="external nofollow ugc" href="http://example.com/author-url/" target="_self" >Test</a></div><div class="wp-block-comment-content"><p>Hello world</p></div></li></ol>',
+			str_replace( array( "\n", "\t" ), '', $block->render() )
+		);
```
Comment (hellofromtonya) [inline tests/phpunit/tests/blocks/renderCommentTemplate.php]:
```suggestion
		$this->assertSame(
			'<ol class="wp-block-comment-template"><li id="comment-' . self::$comment_ids[0] . '" class="comment even thread-odd thread-alt depth-1"><div class="wp-block-comment-author-name"><a rel="external nofollow ugc" href="http://example.com/author-url/" target="_self" >Test</a></div><div class="wp-block-comment-content"><p>Hello world</p></div></li></ol>',
			str_replace( array( "\n", "\t" ), '', $block->render() ),
			'Should not include any unapproved comments after removing filter' 
		);
```

Add failure message to identify which assertion in the failed.

File: tests/phpunit/tests/blocks/renderCommentTemplate.php [hunk 2]
```diff
@@ -348,4 +378,53 @@ function test_build_comment_query_vars_from_block_with_comment_preview() {
...
+				'comment_author_email' => 'unapproved@example.org',
+			);
+		};
+
+		add_filter( 'wp_get_current_commenter', $commenter_filter );
+
+		$this->assertSame(
+			'<ol class="wp-block-comment-template"><li id="comment-' . self::$comment_ids[0] . '" class="comment even thread-odd thread-alt depth-1"><div class="wp-block-comment-author-name"><a rel="external nofollow ugc" href="http://example.com/author-url/" target="_self" >Test</a></div><div class="wp-block-comment-content"><p>Hello world</p></div></li><li id="comment-' . $unapproved_comment[0] . '" class="comment odd alt thread-even depth-1"><div class="wp-block-comment-author-name">Visitor</div><div class="wp-block-comment-content"><p><em class="comment-awaiting-moderation">Your comment is awaiting moderation.</em></p>Hi there! My comment needs moderation.</div></li></ol>',
+			str_replace( array( "\n", "\t" ), '', $block->render() )
+		);
```
Comment (hellofromtonya) [inline tests/phpunit/tests/blocks/renderCommentTemplate.php]:
```suggestion
		$this->assertSame(
			'<ol class="wp-block-comment-template"><li id="comment-' . self::$comment_ids[0] . '" class="comment even thread-odd thread-alt depth-1"><div class="wp-block-comment-author-name"><a rel="external nofollow ugc" href="http://example.com/author-url/" target="_self" >Test</a></div><div class="wp-block-comment-content"><p>Hello world</p></div></li><li id="comment-' . $unapproved_comment[0] . '" class="comment odd alt thread-even depth-1"><div class="wp-block-comment-author-name">Visitor</div><div class="wp-block-comment-content"><p><em class="comment-awaiting-moderation">Your comment is awaiting moderation.</em></p>Hi there! My comment needs moderation.</div></li></ol>',
			str_replace( array( "\n", "\t" ), '', $block->render() ),
			'Should include unapproved comments when filter applied.' 
		);
```

Add failure message to identify which assertion in the failed.

File: tests/phpunit/tests/blocks/renderCommentTemplate.php [hunk 3]
```diff
@@ -348,4 +378,53 @@ function test_build_comment_query_vars_from_block_with_comment_preview() {
...
+			array(
+				'postId' => self::$custom_post->ID,
+			)
+		);
+
+		$commenter_filter = function () {
+			return array(
+				'comment_author_email' => 'unapproved@example.org',
+			);
+		};
```
Comment (hellofromtonya) [inline tests/phpunit/tests/blocks/renderCommentTemplate.php]: Micro-optimization - use static closure for a tiny performance improvement.

```suggestion
		$commenter_filter = static function () {
			return array(
				'comment_author_email' => 'unapproved@example.org',
			);
		};
```

File: tests/phpunit/tests/blocks/renderCommentTemplate.php [hunk 4]
```diff
@@ -307,6 +306,37 @@ function test_rendering_comment_template_nested() {
 		);
 	}
 
+	/**
+	 * Test that line and paragraph breaks are converted to HTML tags in a comment.
+	 */
```
Comment (hellofromtonya) [inline tests/phpunit/tests/blocks/renderCommentTemplate.php]:
```suggestion
	/**
	 * Test that line and paragraph breaks are converted to HTML tags in a comment.
	 *
	 * @ticket 55643
	 */
```

File: tests/phpunit/tests/blocks/renderCommentTemplate.php [hunk 5]
```diff
@@ -348,4 +378,55 @@ function test_build_comment_query_vars_from_block_with_comment_preview() {
 			)
 		);
 	}
+
+	/**
+	 * Test rendering an unapproved comment preview.
+	 */
```
Comment (hellofromtonya) [inline tests/phpunit/tests/blocks/renderCommentTemplate.php]:
```suggestion
	/**
	 * Test rendering an unapproved comment preview.
	 *
	 * @ticket 55643
	 */
```

File: tests/phpunit/tests/blocks/renderCommentTemplate.php [hunk 6]
```diff
@@ -348,4 +380,57 @@ function test_build_comment_query_vars_from_block_with_comment_preview() {
...
+				'comment_author_email' => 'unapproved@example.org',
+			);
+		};
+
+		add_filter( 'wp_get_current_commenter', $commenter_filter );
+
+		$this->assertSame(
+			'<ol class="wp-block-comment-template"><li id="comment-' . self::$comment_ids[0] . '" class="comment even thread-odd thread-alt depth-1"><div class="wp-block-comment-author-name"><a rel="external nofollow ugc" href="http://example.com/author-url/" target="_self" >Test</a></div><div class="wp-block-comment-content"><p>Hello world</p></div></li><li id="comment-' . $unapproved_comment[0] . '" class="comment odd alt thread-even depth-1"><div class="wp-block-comment-author-name">Visitor</div><div class="wp-block-comment-content"><p><em class="comment-awaiting-moderation">Your comment is awaiting moderation.</em></p>Hi there! My comment needs moderation.</div></li></ol>',
+			str_replace( array( "\n", "\t" ), '', $block->render() ),
+			'Should include unapproved comments when filter applied.'
```
Comment (hellofromtonya) [inline tests/phpunit/tests/blocks/renderCommentTemplate.php]:
```suggestion
			'Should include unapproved comments when filter applied'
```

Remove period and whitespace.

File: tests/phpunit/tests/blocks/renderCommentTemplate.php [hunk 7]
```diff
@@ -348,4 +380,57 @@ function test_build_comment_query_vars_from_block_with_comment_preview() {
...
+			'Should include unapproved comments when filter applied.' 
+		);
+
+		remove_filter( 'wp_get_current_commenter', $commenter_filter );
+
+		// Test it again and ensure the unmoderated comment doesn't leak out.
+		$this->assertSame(
+			'<ol class="wp-block-comment-template"><li id="comment-' . self::$comment_ids[0] . '" class="comment even thread-odd thread-alt depth-1"><div class="wp-block-comment-author-name"><a rel="external nofollow ugc" href="http://example.com/author-url/" target="_self" >Test</a></div><div class="wp-block-comment-content"><p>Hello world</p></div></li></ol>',
+			str_replace( array( "\n", "\t" ), '', $block->render() ),
+			'Should not include any unapproved comments after removing filter'
```
Comment (hellofromtonya) [inline tests/phpunit/tests/blocks/renderCommentTemplate.php]:
```suggestion
			'Should not include any unapproved comments after removing filter'
```

Remove end of line whitespace (for phpcs).

**Review by hellofromtonya — APPROVED** [review]
These additional tests are consistent with the other tests in the same class. Ready for commit.

**Comment by hellofromtonya** [comment]
I'll get this PR committed. Then in a new PR, I'll work on resolving the interdependency issues between the tests in this class, ie `even` `odd` class attributes instability.

**Comment by hellofromtonya** [comment]
Committed via changeset https://core.trac.wordpress.org/changeset/53353.

**Pull request closed without merging**